### PR TITLE
✨ Add data to Trello Card Streams

### PR DESCRIPTION
**Pull request by danieldiamond**
## What
Add additional data points to Trello Card Streams e.g. pluginData, members and lists

## How
Extend query params to turn the boolean for these fields to `true`

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? If yes, please make sure to include it here and in any changelogs with the 🚨🚨 emoji
What is the end result perceived by the user?

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-trello/source_trello/source.py [hunk 1]
```diff
@@ -107,7 +107,13 @@ class Cards(ChildStreamMixin, TrelloStream):
...
     parent_stream_class = Boards
     limit = 20000
-    extra_params = {"customFieldItems": "true"}
+    extra_params = {
+        "customFieldItems": "true",
+        "pluginData": "true",
+        "actions_display": "true",
+        "members": "true",
+        "list": "true",
+    }
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-trello/source_trello/source.py:116]: can you comment above or point to documentation on how adding these is helping or what difference it makes

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

**Comment by harshithmullapudi** [comment]
Also you can check this doc here https://docs.airbyte.io/connector-development#4.-publish-the-connector and do the needy before we publish this

File: airbyte-integrations/connectors/source-trello/source_trello/source.py (hunk 1 above)
Comment (danieldiamond) [reply airbyte-integrations/connectors/source-trello/source_trello/source.py:116]: the documentation is provided in the function docs. it points to the exact API thats being used, where the documentation for these `extra_params` is detailed. The param names also are very self-explanatory

**Comment by danieldiamond** [comment]
@harshithmullapudi the docs for the link you provided above state:
> Typically this will be handled as part of code review by an Airbyter. There is a section below on what steps are needed for publishing a connector and will mostly be used by Airbyte employees publishing the connector.

that is not very helpful. also i'm not sure what you mean by "do the needy before we publish this"

**Comment by harshithmullapudi** [comment]
Hey @danieldiamond my bad can you check this https://docs.airbyte.io/connector-development#publishing-a-connector

**Comment by harshithmullapudi** [comment]
@danieldiamond can you run the integration tests in local and share the screenshots here ? Is it possible ?

**Comment by danieldiamond** [comment]
```
➜ python -m pytest unit_tests
Test session starts (platform: darwin, Python 3.9.9, pytest 6.2.5, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/danieldiamond/repos/airbyte, configfile: pytest.ini
plugins: sugar-0.9.4, requests-mock-1.9.3, mock-3.6.1, timeout-1.4.2
collecting ...
 airbyte-integrations/connectors/source-trello/unit_tests/test_control_rate_limit.py::test_with_load ✓           8% ▉
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_cursor_field ✓      17% █▋
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_get_updated_state ✓ 25% ██▌
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_stream_slices ✓     33% ███▍
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_supports_incremental ✓42% ████▎
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_source_defined_cursor ✓50% █████
 airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py::test_stream_checkpoint_interval ✓58% █████▉
 airbyte-integrations/connectors/source-trello/unit_tests/test_source.py::test_streams ✓                        67% ██████▋
 airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_request_params ✓                75% ███████▌
 airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_next_page_token ✓               83% ████████▍ {"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}

 airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_boards_stream ✓                 92% █████████▎{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Sleep 0 seconds based on load conditions."}}

 airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_cards_stream ✓                 100% ██████████
====================================================== warnings summary ======================================================
airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py: 6 warnings
airbyte-integrations/connectors/source-trello/unit_tests/test_source.py: 6 warnings
airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py: 11 warnings
  /Users/danieldiamond/.virtualenvs/trello/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:37: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator = NoAuth()

airbyte-integrations/connectors/source-trello/unit_tests/test_incremental_streams.py: 6 warnings
airbyte-integrations/connectors/source-trello/unit_tests/test_source.py: 6 warnings
airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py: 13 warnings
  /Users/danieldiamond/.virtualenvs/trello/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

airbyte-integrations/connectors/source-trello/unit_tests/test_source.py::test_streams
  /Users/danieldiamond/repos/airbyte/airbyte-integrations/connectors/source-trello/source_trello/source.py:201: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return TrelloAuthenticator(token=token, key=key)

airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_boards_stream
  /Users/danieldiamond/repos/airbyte/airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py:44: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    config = {"authenticator": NoAuth(), "start_date": "2021-02-11T08:35:49.540Z"}

airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py::test_cards_stream
  /Users/danieldiamond/repos/airbyte/airbyte-integrations/connectors/source-trello/unit_tests/test_streams.py:79: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    config = {"authenticator": NoAuth(), "start_date": "2021-02-11T08:35:49.540Z"}

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (0.07s):
      12 passed
```

**Comment by harshithmullapudi** [comment]
Awesome. @danieldiamond  thanks a lot.

**Pull request merged**